service/src/main/java/bio/terra/workspace/db/GrantDao.java

```diff
@@ -0,0 +1,226 @@
+    SELECT grant_id, workspace_id FROM temporary_grant WHERE expire_time < :current_time AND revoke_flight_id IS NULL
+    """;
+
+  private static final RowMapper<ExpiredGrant> GRANT_ID_ROW_MAPPER =
+      (rs, rowNum) ->
+          new ExpiredGrant(
+              UUID.fromString(rs.getString("grant_id")),
+              UUID.fromString(rs.getString("workspace_id")));
+
+  /** Update to lock a grant */
+  private static final String LOCK_GRANT_SQL =
+      """
+    UPDATE temporary_grant SET revoke_flight_id = :flight_id
+    WHERE grant_id = :grant_id AND revoke_flight_id IS NULL
+    """;
+
+  /** Update to unlock a grant */
+  private static final String UNLOCK_GRANT_SQL =
+      """
+    UPDATE temporary_grant SET revoke_flight_id = NULL
+    WHERE grant_id = :grant_id and revoke_flight_id = :flight_id
+    """;
+
+  /** Delete and implicitly unlock a grant */
+  private static final String DELETE_GRANT_SQL =
+      """
+    DELETE FROM temporary_grant WHERE grant_id = :grant_id and revoke_flight_id = :flight_id
+    """;
+  /** Query and mapper to retrieve a grant into GrantData */
+  private static final String GET_GRANT_SQL =
+      """
+  SELECT grant_id, workspace_id, user_member, petsa_member, grant_type, resource_id, role, create_time, expire_time
+  FROM temporary_grant
+  WHERE grant_id = :grant_id
+  """;
+
+  private static final RowMapper<GrantData> GRANT_DATA_ROW_MAPPER =
+      (rs, rowNum) ->
+          new GrantData(
+              UUID.fromString(rs.getString("grant_id")),
+              UUID.fromString(rs.getString("workspace_id")),
+              rs.getString("user_member"),
+              rs.getString("petsa_member"),
+              GrantType.fromDb(rs.getString("grant_type")),
+              Optional.ofNullable(rs.getString("resource_id")).map(UUID::fromString).orElse(null),
+              rs.getString("role"),
+              OffsetDateTime.ofInstant(
+                  rs.getTimestamp("create_time").toInstant(), ZoneId.of("UTC")),
+              OffsetDateTime.ofInstant(
+                  rs.getTimestamp("expire_time").toInstant(), ZoneId.of("UTC")));
+
+  private static final String INSERT_GRANT_SQL =
+      """
+    INSERT INTO temporary_grant
+    (grant_id, workspace_id, user_member, petsa_member, grant_type,
+     resource_id, role, create_time, expire_time)
+    VALUES
+    (:grant_id, :workspace_id, :user_member, :petsa_member, :grant_type,
+     :resource_id, :role, :create_time, :expire_time)
+    """;
+
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  public GrantDao(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  /** @return expired grants that are not already being revoked */
+  @ReadTransaction
+  public List<ExpiredGrant> getExpiredGrants() {
+    var params =
+        new MapSqlParameterSource()
+            .addValue(
+                "current_time",
+                Timestamp.valueOf(
+                    OffsetDateTime.now(ZoneId.of("UTC"))
```

Comment:
I missed this before.

You introduced the use of OffsetDateTime in the activity log and created_date stuff. I thought you needed it because Instant didn't give you timezone, so I followed suit in the time handling for grants. But now I see that Instant is a UTC time. Why not use Instant everywhere instead of OffsetDateTime?

We use Instant elsewhere (cronjob for one) and I don't need anything different than that, so I think I will change mine to just use simple Instant.